okhttp-android/src/main/kotlin/okhttp3/android/AndroidAsyncDns.kt

```diff
@@ -40,38 +40,47 @@ class AndroidAsyncDns(
   private val network: Network? = null,
 ) : AsyncDns {
   @RequiresApi(Build.VERSION_CODES.Q)
-  private val resolver = DnsResolver.getInstance()
+  internal val resolver = DnsResolver.getInstance()
   private val executor = Executors.newSingleThreadExecutor()
 
   override fun query(
     hostname: String,
     callback: AsyncDns.Callback,
   ) {
-    resolver.query(
-      network,
-      hostname,
-      dnsClass.type,
-      DnsResolver.FLAG_EMPTY,
-      executor,
-      null,
-      object : DnsResolver.Callback<List<InetAddress>> {
-        override fun onAnswer(
-          addresses: List<InetAddress>,
-          rCode: Int,
-        ) {
-          callback.onResponse(hostname, addresses)
-        }
+    try {
+      resolver.query(
+        network,
+        hostname,
+        dnsClass.type,
+        DnsResolver.FLAG_EMPTY,
+        executor,
+        null,
+        object : DnsResolver.Callback<List<InetAddress>> {
+          override fun onAnswer(
+            addresses: List<InetAddress>,
+            rCode: Int,
+          ) {
+            callback.onResponse(hostname, addresses)
+          }
 
-        override fun onError(e: DnsResolver.DnsException) {
-          callback.onFailure(
-            hostname,
-            UnknownHostException(e.message).apply {
-              initCause(e)
-            },
-          )
-        }
-      },
-    )
+          override fun onError(e: DnsResolver.DnsException) {
+            callback.onFailure(
+              hostname,
+              UnknownHostException(e.message).apply {
+                initCause(e)
+              },
+            )
+          }
+        },
+      )
+    } catch (e: Exception) {
```

Comment:
Any idea what kinds of exceptions we’ll be potentially hiding here? We’re attributing an arbitrary failure like an IllegalStateException as an unknown host

Reply:
I saw from the attached but report it’s gonna happen when the DNS isn't available.

```
IllegalArgumentException: Network.fromNetworkHandle refusing to instantiate NETID_UNSET Network.
```

I agree with this PR, we’ve got to be resilient to this API that doesn’t offer us a ton of good options.

Reply:
I'll add a comment.

Reply:
https://github.com/square/okhttp/pull/8322/files